File: search-application/src/main/java/com/example/search/application/synonym/service/DeleteSynonymGroupService.java
package com.example.search.application.synonym.service;

import com.example.search.application.synonym.port.in.DeleteSynonymGroupUseCase;
import com.example.search.application.synonym.port.out.SynonymGroupRepository;
import com.example.search.domain.synonym.SynonymGroupId;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Slf4j
public class DeleteSynonymGroupService implements DeleteSynonymGroupUseCase {

    private final SynonymGroupRepository repository;

    @Override
    @Transactional
    public void delete(SynonymGroupId id) {
        repository.deleteById(id);
        log.info("synonym 그룹 삭제 id={}", id.value());
    }
}

File: search-application/src/main/kotlin/com/example/search/application/synonym/service/DeleteSynonymGroupService.kt
package com.example.search.application.synonym.service

import com.example.search.application.synonym.port.`in`.DeleteSynonymGroupUseCase
import com.example.search.application.synonym.port.out.SynonymGroupRepository
import com.example.search.domain.synonym.SynonymGroupId
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class DeleteSynonymGroupService(
    private val repository: SynonymGroupRepository
) : DeleteSynonymGroupUseCase {

    @Transactional
    override fun delete(id: SynonymGroupId) {
        repository.deleteById(id)
        log.info("synonym 그룹 삭제 id={}", id.value)
    }

    companion object {
        private val log = LoggerFactory.getLogger(DeleteSynonymGroupService::class.java)
    }
}
